### PR TITLE
feat(route): add route of obsidian publish

### PR DESCRIPTION
**Pull request by Xy2002**
## Involved Issue / 该 PR 相关 Issue

Close #

## Example for the Proposed Route(s) / 路由地址示例

```routes
/obsidian/publish/marshallontheroad
```

## New RSS Route Checklist / 新 RSS 路由检查表
  
- [x] New Route / 新的路由
  - [x] Follows [Script Standard](https://docs.rsshub.app/joinus/advanced/script-standard) / 跟随 [路由规范](https://docs.rsshub.app/zh/joinus/advanced/script-standard)
- [ ] Anti-bot or rate limit / 反爬/频率限制
  - [ ] If yes, do your code reflect this sign? / 如果有, 是否有对应的措施?
- [x] [Date and time](https://docs.rsshub.app/joinus/advanced/pub-date) / [日期和时间](https://docs.rsshub.app/zh/joinus/advanced/pub-date)
  - [x] Parsed / 可以解析
  - [ ] Correct time zone / 时区正确
- [ ] New package added / 添加了新的包
- [ ] `Puppeteer`

## Note / 说明

对于Obsidian Publish，需要Vault的作者给每一篇md的文档属性添加相关属性，否则作用不大

### CODE REVIEW

File: lib/routes/obsidian/publish.ts [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        const post = preloadCacheResponse[postKey];
+        if (!post) {
+            continue;
+        }
+        const item = post?.frontmatter || {};
+        if (!('title' in item)) {
+            item.title = getTitle(postKey);
+        }
+        items.push(item);
+    }
```
Comment (TonyRL) [inline lib/routes/obsidian/publish.ts:80]:
```ts
    const items: DataItem[] = Object.entries(preloadCacheResponse)
        .map(([postKey, post]) => {
            if (!post) {
                return null;
            }
            const item = post?.frontmatter || {};
            if (!('title' in item)) {
                item.title = getTitle(postKey);
            }
            return item;
        })
        .filter(Boolean);
```

**Review by TonyRL — COMMENTED** [review]
Missing `link` and `pubDate`

File: lib/routes/obsidian/publish.ts [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+            preloadCacheUrl = preloadCacheUrlMatch[1];
+            break;
+        }
+    }
+
+    let preloadCacheResponse;
+    try {
+        preloadCacheResponse = await ofetch(preloadCacheUrl, {
+            headers: {
+                'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/122.0.0.0 Safari/537.36',
```
[the remark this answers is not in the document]
Comment (Xy2002) [reply lib/routes/obsidian/publish.ts:60]: Sorry, I didn't know this attr existed before, I just realized it when I looked at someone else's code, I'll replace it with the ua provided by config

File: lib/routes/obsidian/publish.ts (hunk 1 above)
Comment (Xy2002) [reply lib/routes/obsidian/publish.ts]: Thank you very much for your CR, I learned something new :)

File: lib/routes/obsidian/publish.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
+import { Route, DataItem } from '@/types';
+import { load } from 'cheerio';
+import { ofetch } from 'ofetch';
```
Comment (TonyRL) [inline lib/routes/obsidian/publish.ts:3]:
```suggestion
import ofetch from '@/utils/ofetch';
```

**Pull request merged**